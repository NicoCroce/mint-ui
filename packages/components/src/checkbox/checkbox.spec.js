import * as React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import Checkbox from "./index";

afterEach(cleanup);

const CheckboxState = () => {
	const [checked, setChecked] = React.useState(false);
	const handleChange = val => {
		setChecked(val);
	};
	return <Checkbox value={checked} onChange={handleChange} />;
};

describe("Checkbox", () => {
	const { getByLabelText, asFragment } = render(<CheckboxState />);
	test("changes checkbox state", () => {
		const checkbox = getByLabelText("tm-checkbox");
		fireEvent.click(checkbox, { value: true });
		expect(checkbox.value).toBeTruthy();
	});
	test("renders checkbox", () => {
		expect(asFragment()).toMatchSnapshot();
	});
});
