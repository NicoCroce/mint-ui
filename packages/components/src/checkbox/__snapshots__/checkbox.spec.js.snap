// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders checkbox 1`] = `<DocumentFragment />`;
